features/users/controller.js

```diff
@@ -0,0 +1,158 @@
+"use strict";
+var debugEnabled = require("../../config").debug;
+var log = require("color-logs")(debugEnabled, debugEnabled, __filename);
+var Chance = new require('chance')();
+
+var User = require('./model');
+var Friendship = require('./friendship.service');
+var BaseApi = require("../../api");
+
+var mocks = require('../../test-helpers/_mocks');
+
+class UsersController extends BaseApi {
+
+  async getUserById(id) {
+    try {
+      let user = await User.findById(id).select(['-friends']);
+      let friends = await Friendship.getFriends(id);
+      return {
+        user: user,
+        friends: friends
+      };
+    } catch(ex) {
+      throw ex;
+    }
+  }
+
+  getUser(req, res) {
+    let id = req.params.id;
+    log.info("looking for user id " + id);
+    this.getUserById(id)
+      .then(model => this.success(res, model))
+      .catch(err => this.exception(res, err));
+  }
+
+  async getUsers(req, res) {
+    await User.find()
+      .select(['-friends'])
+      .sort({ "name": 1 })
+      .exec( (err, models) => {
+        if (err) {
+          log.error(err);
+          return this.exception(res, err);
+        }
+        return this.success(res, models);
+      });
+  };
+
+  async getMe(req, res) {
+    let data = this.getToken(req);
+    log.info(data);
+
+    try {
+      let user = await this.getUserById(data.user_id);
+      if (!user) {
+        return this.fail(res, 401, "User not found!");
+      }
+
+      return this.success(res, user);
+    } catch(ex) {
+      return this.exception(res, ex);
+    }
+  }
+
+  createUser(req, res) {
+    var model = new User(req.body);
+    return new Promise((resolve, reject) => {
+      model.save((err, model) => {
+        if (err) {
+          log.error("save user error", err);
+          this.exception(res, err);
+          return reject(err);
+        }
+        this.success(res, model);
+        return resolve(model);
+      });
+    })
+  }
+
+  async meet(req, res) {
+    let user = req.params.id;
+    let friend = req.params.friend;
+    try {
+      await Friendship.makeFriend(user, friend);
+      let response = await this.getUserById(user);
+      return this.success(res, response);
+    } catch(ex) {
+      return this.exception(res, ex);
+    }
+  }
+  async break(req, res) {
+    let user = req.params.id;
+    let friend = req.params.friend;
+    log.info(user + " breaking up with "+ friend);
+    try {
+      await Friendship.breakUp(user, friend);
+      let response = await this.getUserById(user);
+      return this.success(res, response);
+    } catch(ex) {
+      return this.exception(res, ex);
+    }
+  }
+
+  updateUser(req, res) {
+    var id = req.params.id;
+    let data = this.getToken(req);
+    if(!data || data.user_id != id) return this.error(res, 401, "Denied");
+
+    User.findById(id)
+      .exec((err, model) => {
+        let data = req.body;
+        if(data.name) model.name = req.body.name;
```

Comment:
Can you shrink the bunch of if-statements? If we have 30 fields in the model, we have to repeat the if-statement 30 times. Frankly, it's not a good smell.